internal/generate/olm-catalog/package_manifest.go

```diff
@@ -0,0 +1,192 @@
+// manifest and modifies it based on values set in s.
+func (g pkgGenerator) generate() (map[string][]byte, error) {
+	fileName := getPkgFileName(g.OperatorName)
+	path := filepath.Join(g.Inputs[ManifestsDirKey], fileName)
+	pkg := registry.PackageManifest{}
+	if _, err := os.Stat(path); err == nil {
+		b, err := ioutil.ReadFile(path)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read package manifest %s: %v", path, err)
+		}
+		if err = yaml.Unmarshal(b, &pkg); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal package manifest %s: %v", path, err)
+		}
+		if err := registryutil.ValidatePackageManifest(&pkg); err != nil {
+			return nil, fmt.Errorf("error validating package manifest %s: %v", pkg.PackageName, err)
+		}
+	} else if os.IsNotExist(err) {
+		pkg = newPackageManifest(g.OperatorName, g.channel, g.csvVersion)
+	} else {
+		return nil, fmt.Errorf("error reading package manifest %s: %v", path, err)
+	}
+
+	g.setChannels(&pkg)
+	sort.Slice(pkg.Channels, func(i int, j int) bool {
+		return pkg.Channels[i].Name < pkg.Channels[j].Name
+	})
+
+	b, err := yaml.Marshal(pkg)
+	if err != nil {
+		return nil, err
+	}
+	fileMap := map[string][]byte{
+		fileName: b,
+	}
+	return fileMap, nil
+}
+
+func newPackageManifest(operatorName, channelName, version string) registry.PackageManifest {
+	// Take the current CSV version to be the "alpha" channel, as an operator
+	// should only be designated anything more stable than "alpha" by a human.
+	channel := "alpha"
+	if channelName != "" {
+		channel = channelName
+	}
+	lowerOperatorName := strings.ToLower(operatorName)
+	pkg := registry.PackageManifest{
+		PackageName: lowerOperatorName,
+		Channels: []registry.PackageChannel{
+			{Name: channel, CurrentCSVName: getCSVName(lowerOperatorName, version)},
+		},
+		DefaultChannelName: channel,
+	}
+	return pkg
+}
+
+// setChannels checks for duplicate channels in pkg and sets the default
+// channel if possible.
+func (g pkgGenerator) setChannels(pkg *registry.PackageManifest) {
+	if g.channel != "" {
+		channelIdx := -1
+		for i, channel := range pkg.Channels {
+			if channel.Name == g.channel {
+				channelIdx = i
+				break
+			}
+		}
+		lowerOperatorName := strings.ToLower(g.OperatorName)
+		if channelIdx == -1 {
+			pkg.Channels = append(pkg.Channels, registry.PackageChannel{
+				Name:           g.channel,
+				CurrentCSVName: getCSVName(lowerOperatorName, g.csvVersion),
+			})
+		} else {
+			pkg.Channels[channelIdx].CurrentCSVName = getCSVName(lowerOperatorName, g.csvVersion)
+		}
+		// Use g.channel as the default channel if caller has specified it as the
+		// default.
+		if g.channelIsDefault {
+			pkg.DefaultChannelName = g.channel
+		}
+	}
+
+	if pkg.DefaultChannelName == "" {
+		log.Warn("Package manifest default channel is empty and should be set to an existing channel.")
+	} else {
+		defaultExists := false
+		for _, c := range pkg.Channels {
+			if pkg.DefaultChannelName == c.Name {
+				defaultExists = true
```

Comment:
Could we add a break here?
If found is OK. If not should return the error. Am I right?